dev/PushNotifications/PushNotificationChannel.cpp

```diff
@@ -48,18 +51,75 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
 
     winrt::event_token PushNotificationChannel::PushReceived(winrt::Windows::TypedEventHandler<winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel, winrt::Microsoft::Windows::PushNotifications::PushNotificationReceivedEventArgs> handler)
     {
-        return m_channel.PushNotificationReceived([weak_self = get_weak(), handler](auto&&, auto&& args)
+        if (IsPackagedApp())
+        {
+            return m_channel.PushNotificationReceived([weak_self = get_weak(), handler](auto&&, auto&& args)
+            {
+                auto strong = weak_self.get();
+                if (strong)
+                {
+                    handler(*strong, winrt::make<winrt::Microsoft::Windows::PushNotifications::implementation::PushNotificationReceivedEventArgs>(args));
+                };
+            });
+        }
+        else
         {
-            auto strong = weak_self.get();
-            if (strong)
+            if (!m_isRegisteredWithLRP)
             {
-                handler(*strong, winrt::make<winrt::Microsoft::Windows::PushNotifications::implementation::PushNotificationReceivedEventArgs>(args));
-            };
-        });
+                wil::com_ptr<INotificationsLongRunningPlatform> notificationsLongRunningPlatform{
+                    wil::CoCreateInstance<NotificationsLongRunningPlatform, INotificationsLongRunningPlatform>(CLSCTX_LOCAL_SERVER) };
+
+                wchar_t processName[1024];
+                THROW_HR_IF(ERROR_FILE_NOT_FOUND, GetModuleFileNameExW(GetCurrentProcess(), NULL, processName, sizeof(processName) / sizeof(processName[0])) == 0);
+
+                THROW_IF_FAILED(notificationsLongRunningPlatform->RegisterForegroundActivator(this, processName));
+
+                m_isRegisteredWithLRP = true;
+            }
+            return m_foregroundHandlers.add(handler);
+        }
     }
 
     void PushNotificationChannel::PushReceived(winrt::event_token const& token) noexcept
     {
-        m_channel.PushNotificationReceived(token);
+        if (IsPackagedApp())
```

Comment:
Same issue as above here. This check is incorrect.

Reply:
Function needs to be renamed.